source/faq.txt

```diff
@@ -1,17 +1,101 @@
+monitoring sockets. If the application uses the default setting for
+``maxPoolSize`` and only queries the primary (default) node, then
+there can be at most ``106`` total connections in the connection pool. If the
+application uses a :ref:`read preference <read-preference>` to query the
+secondary nodes, those connection pools grow and there can be
+``306`` total connections.
+
+To support high numbers of concurrent MongoDB requests
+within one process, you can increase ``maxPoolSize``.
+
+Connection pools are rate-limited. The ``maxConnecting`` option
+determines the number of connections that the pool can create in
+parallel at any time. For example, if the value of ``maxConnecting`` is
+``2``, the third request that attempts to concurrently check out a
+connection succeeds only when one the following cases occurs:
+
+- The connection pool finishes creating a connection and there are fewer
+  than ``maxPoolSize`` connections in the pool.
+- An existing connection is checked back into the pool.
+- The driver's ability to reuse existing connections improves due to
+  rate-limits on connection creation.
+
+You can set the minimum number of concurrent connections to
+each server with the ``minPoolSize`` option, which defaults to ``0``.
+The driver initializes the connection pool with this number of sockets. If
+sockets are closed, causing the total number
+of sockets (both in use and idle) to drop below the minimum, more
+sockets are opened until the minimum is reached.
+
+You can set the maximum number of milliseconds that a connection can
+remain idle in the pool by setting the ``maxIdleTimeMS`` option.
+Once a connection has been idle for ``maxIdleTimeMS``, the connection
+pool removes and replaces it. This option defaults to ``0`` (no limit).
+
+The following default configuration for a ``MongoClient`` works for most
+applications:
+
+.. code-block:: js
+   
+   const client = new MongoClient("<connection string>");
+
+``MongoClient`` supports multiple concurrent requests. For each process,
+create a client and reuse it for all operations in a process. This
+practice is more efficient than creating a client for each request.
+
+The driver does not limit the number of requests that
+can wait for sockets to become available, and it is the application's
+responsibility to limit the size of its pool to bound queuing
+during a load spike. Requests wait for the amount of time specified in
+the ``waitQueueTimeoutMS`` option, which defaults to ``0`` (no limit).
+
+A request that waits more than the length of time defined by
+``waitQueueTimeoutMS`` for a socket raises a connection error. Use this
+option if it is more important to bound the duration of operations
+during a load spike than it is to complete every operation.
+
+When ``MongoClient.close()`` is called by any request, the driver
+closes all idle sockets and closes all sockets that are in
+use as they are returned to the pool.
```

Comment:
I would expand this sentence just a bit because it captures an important detail:
"closes all sockets that are in use **as they are returned** to the pool." Meaning we won't interrupt in progress operations. This has a unique downside for Node.js users that their process won't exit if say a slowQuery `find` is still pending results. 

Related issue https://jira.mongodb.org/browse/NODE-3263
We have no clean solution yet because there is risk to an API that cancels operations client side without making clear the impacts (think: unacknowledged write, without actually asking for it to be unacknowledged). There is a future where client-side operation timeout may provide an avenue for users to put a global cap on operation time, so that would provide an avenue for pending operations to be closed at least on _some_ kind of deadline.

TL;DR: I just think the "_as returned_" should be hoisted more in the phrasing b/c of its significance to our users.